Examples/UtiliKit-iOSExample/Container/WipeTransitionAnimator.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  WipeTransitionAnimator.swift
+//  UtiliKit
+//
+//  Copyright © 2018 Bottle Rocket Studios. All rights reserved.
+//
+
+import UIKit
+import UtiliKit
+
+public class WipeTransitionAnimator: NSObject {
+    
+    public enum Direction {
+        case leftToRight
+        case rightToLeft
+    }
+    
+    //MARK: Properties
+    var transitionDirection: Direction
+    private var interruptibleAnimator: UIViewPropertyAnimator?
+    private let duration = 0.3
+    
+    //MARK: Initializers
+    public init(direction: Direction) {
+        self.transitionDirection = direction
+        super.init()
+    }
+    
+    public convenience init(startIndex: Int, endIndex: Int) {
+        self.init(direction: startIndex < endIndex ? .rightToLeft : .leftToRight)
+    }
+    
+    // MARK: Interface
+    func configure(forStartIndex startIndex: Int, endIndex: Int) {
+        transitionDirection = startIndex < endIndex ? .rightToLeft : .leftToRight
+    }
+}
+
+//MARK: ContainerViewControllerAnimatedTransitioning
+extension WipeTransitionAnimator: ContainerViewControllerAnimatedTransitioning {
+    
+    public func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return duration
+    }
+    
+    public func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        interruptibleAnimator?.startAnimation()
+    }
+    
+    public func interruptibleAnimator(using transitionContext: UIViewControllerContextTransitioning) -> UIViewImplicitlyAnimating {
+        if let animator = interruptibleAnimator {
+            return animator
+        }
+        
+        guard let destination = transitionContext.viewController(forKey: .to), let source = transitionContext.viewController(forKey: .from) else {
+            fatalError("The context is improperly configured - require both a source and destination.")
+        }
+
+        configureInitialState(for: destination, with: transitionContext)
+        let timingParameters = UICubicTimingParameters(animationCurve: .easeInOut)
+        let propertyAnimator = UIViewPropertyAnimator(duration: transitionDuration(using: transitionContext), timingParameters: timingParameters)
+        propertyAnimator.addAnimations { [unowned self] in
```

Comment:
what is people's opinion of using `unowned self` in the open source projects?

Reply:
I have the same opinion as I do in client projects - if you can reason that the self being referenced as unowned is the object retaining the closure doing the referencing then you are safe to use it as you can guarantee it won't be deallocated when that closure executes.

Reply:
I agree with Will on this one. I'm fine with it.

Reply:
Personally, I'm not a fan of `unowned`. As a reader, I either have to put all of my trust in the author that they knew what they're doing or spend extra time to verify they did it correctly.

In this case, I'm not even sure if I'm able to quickly understand that this is a "safe" use of `unowned`. The closure is owned by the `propertyAnimator`, which ends up being returned from this function so someone else (whoever interacts with `WipeTransitionAnimator` through the `UIViewControllerAnimatedTransitioning` protocol) must be retaining it in a way that's safe.

I get that `unowned` might be convenient sometimes when you need to access properties or functions on `self` from inside the closure, but in this example, it doesn't matter. Instead, it feels like we are forced to deal with all the extra cognitive load (or potentially a bug/crash) just to avoid doing some optional chaining.
```swift
        propertyAnimator.addAnimations { [weak self] in
            self?.configureFinalState(forSource: source, destination: destination, context: transitionContext)
        }
```

Since this is "example" code, I suppose it's _okay_ to leave it as `unowned`, but even from a best-practice standpoint I'd make the argument that encouraging `weak` would be better.

Reply:
The property animator is also retained by the `UIViewControllerAnimatedTransitioning` object creating it. 

I will make these `weak`.